### ISSUE TITLE
Endpoint to get current statistics

### ISSUE THREAD

**Issue by iwoork** [findthecluster/api-app#18]
After the user has completed the form and presented the results page, it will be great to show the existing data about the postal/zip code he is in.

For example, if the user inputs M4A2B1 as post code, there should be an API that returns the data for:
- Confimed
- Deaths
- Recovered

Suggestion:
`GET /statistics?post_code=M4A2B1&date=2020-03-20`

parameters:
- post code 
- date

Response:
```
{
  city: 'New York,
  confirmed: 123,
  deaths: 0,
  recovered: 0,
  updateDate: '2020-03-20'
}
```

**Comment by iwoork** [comment]
<img width="1293" alt="Screen Shot 2020-04-18 at 12 19 32 AM" src="https://user-images.githubusercontent.com/1687468/79627911-52618400-810a-11ea-8d7d-0a418a1a7b5e.png">

**Comment by rashnil-git** [comment]
we don't have current deaths and confirmed. we can find that data set online and store in db and return the results as response of the same `/stats` endpoint

**Comment by rashnil-git** [comment]
we need to get that data by zip code first.

**Comment by iwoork** [comment]
I agree. right now i'm hitting a public endpoint first https://github.com/summyfeb12/COVID-19-JHU-Data-API and filtering by a hardcoded string:

https://github.com/checkifcovid/survey-app/blob/master/pages/api/statistics.js

**Comment by rashnil-git** [comment]
@iwoork : We have state level and county level data online for US. I was thinking if we can change the zip code to two options `State` and `county` in UI then we should be able to retrieve the latest confirmed cases and deaths at county level??

**Comment by rashnil-git** [comment]
I feel not sure if we will have the post code level granularity at this stage.

**Comment by rashnil-git** [comment]
we can make zip code field as optional. We can use that information to track in our DB, but collect state and county from user to fetch latest confirmed cases and death counts as response to stats api. Let me know ?

**Comment by rashnil-git** [comment]
I looked at some data. Here is my recommendations:
1. Add State and County as well in addition to zip code.
2. I found a service for US zips to get zip level stats on confirmed, deaths and recovered counts.

I am going build this into the /stats endpoint:
So, stats api will return
```
{
  statusCode: 200,
  Diagnosis: 'Positive',
  probability: 0.8,
  msg: 'Probability Calculated'
  statistics: 'zip'   ---zip/county/state based on available info in DB
  cases: 245
  deaths: 12
  recovered: 233
}
```
I need the survey app to pass zip code, state and county. if zip not available, then will look for county or state level info and return.

I will build this in coming weekend. Let me know before then.

**Comment by iwoork** [comment]
Here's my thoughts:

1. Maintain capturing just zip/postal code as part of the form. Add UI validation to make sure it is a valid postal/zip code.
2. Post capture (either through a separate process or when querying data, create a map of the postcode to state/county level and save it either same row or separate table.

I feel this makes it easier to users so they dont have to search for their state and county. I was able to geocode a postcode and it looks much richer to work with. The cost is expensive though at $5 for 1000 requests.

See sample response.
<img width="437" alt="Screen Shot 2020-04-25 at 12 08 23 AM" src="https://user-images.githubusercontent.com/1687468/80270864-b4c9ff80-8689-11ea-8f7c-a4b94f5dbb00.png">

what do you think?

**Comment by iwoork** [comment]
And perhaps separate the endpoints?
`/diagnose` to use for getting the:
```
{
  probability: 0.8,
  diagnosis: 'positive', // I'm not comfortable adding diagnosis. i feel we just give them probability
  accuracy: 0.4,
  postcode: 10001,
  symptoms: {
    fever: true,
    cough: true
  }
}
```
another endpoint `/statistics` to just get the cases data:
```
{
  cases: 1234,
  deaths: 13,
  recovered: 1000
}
```

**Comment by rashnil-git** [comment]
Few points:
1. I feel as of now, we may not need to pay for the service. There are lot of data available for valid postal codes across the world. that data can be used to validate zip codes on website.
2. UI should be as light as possible. We should not create any map in UI, rather post code should be sent to backend for processing and storage or creating mappings.
3. I feel the current endpoint `/stats` is good to handle both the above cases. We can separate them later when we see load on the system. Currently, there are almost 0 requests coming, I feel this would be over engineering. Lets try to build the basic infra and then expand as per need? 

So, `/stats` endpoint will return :
```
{
  probability: 0.8,
  diagnosis: 'positive',  // you can filter in UI if you don't want to display. This was a mock. 
  accuracy: 0.4,
  postcode: 10001,
  symptoms: {.  //I am not sure why we need symptoms as response?? Is it counts or whether anyone had that symptom in the post code? Not clear on requirement here. 
    fever: true,
    cough: true
  }, 
  cases: 1234,
  deaths: 13,
  recovered: 1000
}
```

**Comment by iwoork** [comment]
Sounds good! Much simpler :)

**Comment by iwoork** [comment]
Apologies, when i say "map" i mean mapping of zip code to country/state information :)

**Comment by rashnil-git** [comment]
Can you clarify symptoms question above?

**Comment by iwoork** [comment]
ah my bad. Symptoms is to list the symptoms of the user together with the number of counts within the same postcode. It should be a number instead of boolean value:

e.g
```
symptoms: {
  cough: 100
  fever: 23,
  sore_throat: 63
}
```
